src/dividat-driver/senso/websocket.go

```diff
@@ -293,6 +338,12 @@ func (handle *Handle) dispatchCommand(ctx context.Context, log *logrus.Entry, co
 
 		return nil
 
+	} else if command.UpdateFirmware != nil {
+		handle.ProcessFirmwareUpdateRequest(*command.UpdateFirmware, func(msg FirmwareUpdateMessage) {
+			sendMessage(Message{
+				FirmwareUpdateMessage: &msg,
+			})
+		})
```

Comment:
I think we need to track if an update is in progress and ignore commands that come in the meantime.
To do that, we need to process the update command asynchronously (right now all incoming messages are processed synchronously) so we can keep listening to incoming websocket messages and actively ignore them.

Reply:
[Ignore commands while firmware update is in progress](https://github.com/dividat/driver/pull/127/commits/97fd00550e6f31bd8dddbca12c2dbddc9b6b68b7)